python/pyarrow/io.pxi

```diff
@@ -65,45 +66,63 @@ cdef class NativeFile:
 
         def __get__(self):
             # Emulate built-in file modes
-            if self.is_readable and self.is_writeable:
+            if self.is_readable and self.is_writable:
                 return 'rb+'
             elif self.is_readable:
                 return 'rb'
-            elif self.is_writeable:
+            elif self.is_writable:
                 return 'wb'
             else:
                 raise ValueError('File object is malformed, has no mode')
 
+    def readable(self):
+        self._assert_open()
+        return self.is_readable
+
+    def writable(self):
+        self._assert_open()
+        return self.is_writable
+
+    def seekable(self):
+        self._assert_open()
+        return self.is_readable
```

Comment:
Should these be properties?

Reply:
The `seekable`/`readable`/`writable`? No, this is required for the python file interface. See https://docs.python.org/3/library/io.html#io.IOBase

Reply:
Roger, thanks